pkgs/racket-doc/scribblings/reference/for.scrbl

```diff
@@ -716,7 +716,16 @@ instead of @racket[syntax-protect].
 (for/list ([d (in-digits 1138)]) d)
 
 (map in-digits (list 137 216))
-]}
+]
+
+Sequence syntax transformer procedures should not arm their
+output clauses with syntax @tech{dye packs}, though they
+may arm expressions embedded within the clauses. Because of
+this, avoid using @racket[syntax-rules] for
```

Comment:
The Racket Guide for `syntax-protect` mentions `syntax-rules`, `syntax-id-rules`, and `define-syntax-rule` as forms that automatically protect their expansion results (in the stdlib). Should we include all of them here too for consistency? Alternatively, use the phrase "avoid using forms like @racket[syntax-rules]"?

Reply:
That's a good idea. Does

> avoid using forms like `syntax-rules` and `syntax-id-rules`

Sound good enough? I don't think `define-syntax-rule` applies as much because it doesn't return a transformer value like `syntax-rules` and `syntax-id-rules` do.

Reply:
Oh yeah, good point!

Reply:
I have updated it so the pattern-expander docs say "forms like `syntax-rules` and `syntax-id-rules`, and the sequence-syntax docs say "forms like `syntax-rules`".